app/routes/agenda-items.ts

```diff
@@ -65,42 +64,69 @@ const getQuery = ({
   },
 });
 
-interface AgendaItemsRequestInterface {
-  page: {
-    number: number;
-    size: number;
-  };
-  include: string;
-  sort?: string;
-  filter?: {
-    ':or:'?: object;
-    sessions?: {
-      ':gt:planned-start'?: string;
-      ':lt:planned-start'?: string;
-      'governing-body'?: {
-        'is-time-specialization-of'?: {
-          'administrative-unit': {
-            location?: object;
-          };
-        };
-      };
-    };
-  };
+export async function getAgendaItems(
+  context: AgendaItemsRoute | AgendaItemsController,
+  params: AgendaItemsParams,
+  currentPage = 0
+) {
+  const controller =
```

Comment:
**Nitpick** I feel like this method has too much on its shoulders. What about only keep the query call here, and moving out all code related to controller/context?

Reply:
Agreed. I think this can be simplified a lot. It's a bit confusing to follow at the moment. I took the opportunity to experiment with "Resources" a bit, so I'll PR something soon that should improve that a bit.